ethvm/main.cpp

```diff
@@ -182,17 +183,36 @@ int main(int argc, char** argv)
 			transactionDataString = argv[++i];
 		else if (arg == "--code" && i + 1 < argc)
 			contractCodeString = argv[++i];
+		else
+			incoming = arg;
 	}
 
 	VMFactory::setKind(vmKind);
 
-	bytes contractCode;
-	bytes transactionDataCode = fromHex(transactionDataString);
+
+	//Read transaction data field from input
+	bytes code;
+	if (incoming == "--" || incoming.empty())
```

Comment:
I missed that. It should be only `if (incoming == "--")`. In case `incoming` is empty we should assume no input is provided by standard input. Otherwise the program waits for input in cases it is not expected.